lib/posix/Kconfig

```diff
@@ -14,6 +14,7 @@ config	POSIX_MAX_FDS
 
 config POSIX_API
 	bool "POSIX APIs"
+	select NEWLIB_LIBC
```

Comment:
This isn't correct, and as you've discovered has broken a bunch of tests that were passing on platforms that don't provide newlib.

I mean... the commit message is correct but sort of missing the point.  POSIX does indeed depend on a bunch of stuff from the C standard.  But our POSIX **subset** currently provides **some** of the POSIX standard above our minimal libc **subset**.  And that has value, because those features work on platforms that don't provide newlib.

Please yank this bit and file bugs against libc features that are missing from minimal libc, or alternatively I guess you could advocate for making newlib the default everywhere (which would be a ton of work, but possible).

Reply:
> POSIX does indeed depend on a bunch of stuff from the C standard.

No quite, POSIX itself includes definitions and requirements for libc (and they're even so kind as to point where they differ from ANSI C standard).

> But our POSIX **subset** currently provides **some** of the POSIX standard above our minimal libc **subset**.

Let me translate that to a language which might be understood by a wider audience: we have not-really-working POSIX subsystem, and we brag that half of that works with minimal libc. Quite an accomplishment, but can't go to a conference with that, need something better ;-).

> or alternatively I guess you could advocate for making newlib the default everywhere (which would be a ton of work, but possible).

No, I'm vocally against that. But have to conclude that for POSIX compatibility, newlib is a must.

> Please yank this bit and file bugs against libc features that are missing from minimal libc

No, my plan is the opposite - first make POSIX work in no-nonsense way, then look which subset of that we can offer with minimal libc.